mappers/bluetooth_mapper/README.md

```diff
@@ -0,0 +1,318 @@
+ should be executed or we could let it run infinitely. The user can provide the scheduler input through the configuration file or through the MQTT broker at runtime. The following is the structure used by scheduler module:
+
+  ```go
+// Schedule is structure to define a schedule
+type Schedule struct {
+	//Enable indicates whether the scheduler is to be enabled or not
+	Enable bool `yaml:"enable" json:"enable"`
+	// Name is name of the schedule. It should be unique so that a stop-chan
+	// can be made corresponding to name to stop the schedule.
+	Name string `yaml:"name" json:"name"`
+	// Frequency is the time in milliseconds after which this action are to be performed
+	EventFrequency int `yaml:"event-frequency" json:"event-frequency"`
+	//OccurrenceLimit refers to the number of time the action can occur, if it is 0, then the  event will execute infinitely
+	OccurrenceLimit int `yaml:"occurrence-limit" json:"occurrence-limit"`
+	// Actions is list of Actions to be performed in this schedule
+	Actions []string `yaml:"actions"`
+}
+
+//Scheduler structure contains the list of schedules to be scheduled
+type Scheduler struct {
+	Schedules []Schedule `yaml:"schedules" json:"schedules"`
+}
+```
+   
+ 
+ ### Watcher
+ 
+The following are the responsibilities of the watcher :
+
+a) To scan Bluetooth devices and connect to the correct device once it is in range. 
+
+b) Keep a watch on the expected state of the twin-attributes of the device and perform the action(s) to make actual state equal to expected.
+
+c) To report the actual state of twin attributes back to the cloud.
+
+The inputs for the watcher module can be provided through the configuration file or through the MQTT broker at runtime.
+The following is the structure used by watcher module :
+
+```go
+//Watch structure contains the watcher specific configurations
+type Watch struct {
+	Enable     bool        `yaml:"enable" json:"enable"`
+	Attributes []Attribute `yaml:"attributes" json:"attributes"`
+}
+
+//Attribute structure contains the name of the attribute along with the actions to be performed for this attribute
+type Attribute struct {
+	Name    string   `yaml:"name" json:"name"`
+	Actions []string `yaml:"actions" json:"actions"`
+}
+```
+
+
+ ### Data Converter
+ 
+  The responsibility of data-converter is to convert the readings from the bluetooth device into a format understood by KubeEdge and vice-versa.
+  The input for this module can be provided by the user using the configuration file. The user needs to provide different conversion methods for write operations
+  and read operations and as per the requirements of their device.
+ 
+ ### Controller
+ 
+ Controller exposes MQTT API's for CRUD operations for managing Actions, Schedules and  Watchers. The controller is also responsible for starting all the other 
+  modules initially.   
+ 
+ ## Usage
+ 
+ ### Configuration File
```

Comment:
A silly question, what component will consume this configuration file? Please point it out.

Reply:
Done, thanks